atlasaction/action.go

```diff
@@ -833,14 +840,27 @@ func (a *Actions) MonitorSchema(ctx context.Context) error {
 	}); err != nil {
 		return fmt.Errorf("failed to login to Atlas Cloud: %w", err)
 	}
-	res, err := a.Atlas.SchemaInspect(ctx, &atlasexec.SchemaInspectParams{
-		URL:     db.String(),
+	// Get the URL on the indentifier if not provided (needed for the snapshot hash).
+	if id.URL == "" {
+		if id.URL, err =  a.Atlas.SchemaInspect(ctx, &atlasexec.SchemaInspectParams{
+			ConfigURL: config,
+			Env:       env,
+			Format:  `{{ .RedactedURL }}`,
+		}); err != nil {
+			return fmt.Errorf("failed to inspect the schema: %w", err)
+		}
+	}
+	params := &atlasexec.SchemaInspectParams{
 		Schema:  id.Schemas,
 		Exclude: id.Schemas,
 		Format:  `{{ printf "# %s\n%s" .Hash .MarshalHCL }}`,
-	})
-	if err != nil {
-		return fmt.Errorf("failed to inspect the schema: %w", err)
+	}
+	if db.String() != "" {
+		params.URL = db.String()
+	}
+	if config != "" {
+		params.ConfigURL = config
+		params.Env = env
```

Comment:
Don't inspect twice. Just always take the URL from what Atlas returned here.

Reply:
We can always provide the URL with the config, empty value will be ignore from the flag.